### PR TITLE
fix(cdk): cdk depends on a version of netmask impacted by cve-2021-28918

### PR DESCRIPTION
**Pull request by nija-at**
The fix here is to fork and vendor in the npm package proxy-agent.
Further, apply a patch to remove the support for 'pac' protocol.

The implication of this is that the 'pac' protocols are no longer
supported by the proxy feature of the AWS CDK.

This will be reverted once the upstream node packages upgrade
to netmask@2.0.1 or greater.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nija-at** [comment]
### Testing

- [X] Verify that a normal http proxy works as expected
- [X] Verify that an error message is produced if the proxy uses one of the 'pac' protocols
- [ ] Verify installation and proxy functionality
- [ ] Verify npmignore is correctly recognized

File: packages/aws-cdk/lib/init-templates/v1/app/javascript/package.template.json [hunk 1]
```diff
@@ -16,5 +16,8 @@
   },
   "dependencies": {
     "@aws-cdk/core": "%cdk-version%"
+  },
+  "resolutions": {
+    "netmask": "^2.0.1"
```
Comment (RomainMuller) [inline packages/aws-cdk/lib/init-templates/v1/app/javascript/package.template.json:21]: Are those necessary? I mean they'll be shipping with a newer CLI that has a corrected dependency...

File: packages/aws-cdk/package.json [hunk 2]
```diff
@@ -66,6 +67,19 @@
...
   },
+  "comment-dependencies-proxy-agent": {
+    "desc": "The following dependencies are necessary for the 'proxy-agent' fork. See proxy-agent/README.md",
+    "dependencies": [
+      "agent-base",
+      "debug",
+      "http-proxy-agent",
+      "https-proxy-agent",
+      "lru-cache",
+      "pac-proxy-agent",
```
Comment (RomainMuller) [inline packages/aws-cdk/package.json:78]: Well technically - you're removing this one aren't you?

File: packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts [hunk 3]
```diff
@@ -383,7 +383,7 @@ function parseHttpOptions(options: SdkHttpOptions) {
     // https://aws.amazon.com/blogs/developer/using-the-aws-sdk-for-javascript-from-behind-a-proxy/
     debug('Using proxy server: %s', proxyAddress);
     // eslint-disable-next-line @typescript-eslint/no-require-imports
-    const ProxyAgent: any = require('proxy-agent');
+    const ProxyAgent: any = require('../../../proxy-agent');
```
Comment (RomainMuller) [inline packages/aws-cdk/lib/api/aws-auth/sdk-provider.ts:386]: Would recommend making it such that it's:

```suggestion
    const ProxyAgent: any = require('../../../vendor/proxy-agent');
```

**Comment by nija-at** [comment]
superceded by https://github.com/aws/aws-cdk/pull/13914

**Pull request closed without merging**